x-pack/legacy/plugins/maps/public/shared/layers/sources/ems_tms_source/ems_tms_source.js

```diff
@@ -24,7 +26,7 @@ export class EMSTMSSource extends AbstractTMSSource {
   });
   static icon = 'emsApp';
 
-  static createDescriptor(serviceId) {
+  static createDescriptor(serviceId = EMS_TILE_AUTO_ID) {
```

Comment:
I would change the shape of the descriptor, so the id-field does not include the magical `auto` param. Introduce a new property to distinguish.

e.g.

```
sourceDescriptor: {
   mode: [EMS_TILE_MODE.STATIC|EMS_TILE_MODE.AUTOSELECT]
   id: '...', //use this where mode is EMS_TILE_MODE.STATIC. Can leave blank when mode is EMS_TILE_MODE.AUTOSELECT
   type: 'EMS_TMS'
}
```
Use `EMSTmsSource.createDescriptor` to handle polyfilling for older saved-objects where the mode-property is missing.

There's other options to model this addition I'm sure, but as long as it is not overloading the existing `id`-property.